service/build.go

```diff
@@ -48,26 +49,33 @@ func constructDependencies(cfg *config.Config) (*Dependencies, error) {
 	if err != nil {
 		return nil, err
 	}
-	fs, err := buildFileStore(cfg)
+	fs, err := buildFileStore(cfg, HTTPTimeout)
 	if err != nil {
 		return nil, err
 	}
 	d.BlobStore = core.New(fs, ns)
 	return &d, nil
 }
 
-func buildFileStore(cfg *config.Config) (filestore.FileStore, error) {
+func buildFileStore(cfg *config.Config, HTTPTimeout time.Duration) (filestore.FileStore, error) {
 	trueref := true
 
 	sess := session.Must(session.NewSession())
 	creds := credentials.NewStaticCredentials(cfg.S3AccessKey, cfg.S3AccessSecret, "")
 
 	// need a custom transport to support not verifying SSL cert
+	// if you modify the SSL code, be sure to manually test against
+	// a minio instance with a self-signed certificate
 	customTransport := &http.Transport{
 	    TLSClientConfig: &tls.Config{InsecureSkipVerify: cfg.S3DisableSSLVerify},
         }
-	customHTTPClient := &http.Client{Transport: customTransport}
+	customHTTPClient := &http.Client{
+	    Transport:       customTransport,
+            Timeout:         HTTPTimeout }
```

Comment:
ident weirdness

Reply:
I don't know why the Github editor keeps messing up the indentation.  Ideally I should use a real editor but I'm too lazy.

Reply:
> Github editor

You're using that? I knew you were a masochist but my god

Reply:
I'm just trying to make more style problems for you to comment on